src/diffusers/models/unet_2d_blocks.py

```diff
@@ -872,6 +878,9 @@ def custom_forward(*inputs):
 
             output_states += (hidden_states,)
 
+        if additional_residuals is not None:
+            hidden_states += additional_residuals
```

Comment:
@HimariO 
The implementation above is not correct. you also need to update the `output_states` with new hidden_states, so the code should look like this to match original implementation. Without it the results do not match original implementation 
```python
if additional_residuals is not None:
    hidden_states += additional_residuals
    output_states = output_states[:-1] + (hidden_states,)
```